File: leetcode/Medium/2285. Maximum Total Importance of Roads/2285.java
class Solution {
    public long maximumImportance(int n, int[][] roads) {
        long res = 0L;
        long[] freq = new long[n];
        for(int[] r:roads){
            freq[r[0]]++;
            freq[r[1]]++;
        }
        Arrays.sort(freq);
        long i = 1;
        for(long f:freq){
            res += i*f;
            i++;
        }

        return res;
    }
}

// TC: O(nlog(n))
// SC: O(n)

File: leetcode/Medium/2285. Maximum Total Importance of Roads/2285.cpp
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        long long res = 0, cnt = 1;
        vector<long long> freq(n, 0);
        
        for(auto& r:roads){
            freq[r[0]]++;
            freq[r[1]]++;
        }

        sort(freq.begin(), freq.end());
        for(auto& f:freq){
            res += cnt * f;
            cnt++;
        }

        return res;
    }
};

// TC: O(nlog(n))
// SC: O(n)
